### ISSUE TITLE
Create deterministic shuffling

### ISSUE THREAD

**Issue by joecummings** [pytorch/torchtune#55]
It is NOT deterministic unless we let the user provides a seed (https://fburl.com/trfvxkon) and use that in combination with epoch count to set the generator.  In the dataloader initialization, we should use https://fburl.com/ghrgxgjo (DistributedSampler) with seed provided by the user. It also requires epoch count to be set using set_epoch.

My suggestion would be to retain the changes in this PR, and I will address this issue in a separate PR as part of determinism/shuffling work.

_Originally posted by @gokulavasan in https://github.com/pytorch-labs/torchtune/pull/49#discussion_r1418268406_
